packages/EasyDecision/src/Decisions/DecisionFactory.php

```diff
@@ -201,4 +209,11 @@ private function instantiateDecision(string $decisionType): DecisionInterface
             DecisionInterface::class
         ));
     }
+
+    public function createByName(string $name): DecisionInterface
+    {
+        $decision = $this->mappingProvider->getDecisionType($name);
+
+        return $this->configureDecision(new $decision());
```

Comment:
Can you please pass the `$name` as a constructor parameter to the new `$decision` and we're all good to go! 😄